python/cudf/cudf/_lib/copying.pyx

```diff
@@ -656,3 +656,27 @@ def get_element(Column input_column, size_type index):
         )
 
     return Scalar.from_unique_ptr(move(c_output))
+
+
+def sample(Table input, size_type n,
+           bool replace, int seed, bool keep_index=True):
+    cdef table_view tbl_view = (
+        input.view() if keep_index else input.data_view()
+    )
+
+    cdef unique_ptr[table] c_output
+    with nogil:
+        c_output = move(
+            cpp_copying.sample(tbl_view, n, replace, seed)
+        )
+
+    return Table.from_unique_ptr(
+        move(c_output),
+        column_names=input._column_names,
+        index_names=(
+            None if (
+                input._index is None)
+            or keep_index is False
+            else input._index_names
+        )
```

Comment:
I think we can update this to something like:
```
index_names=(
    None if keep_index is False
    else input._index_names
)
```

IIRC input._index_names automatically handles checking if `._index` is None and returns accordingly.